plugin/metrics/prometheus/options.go

```diff
@@ -64,7 +64,7 @@ func NewOptions(primaryNamespace string) *Options {
 		ServerURL:      defaultServerURL,
 		ConnectTimeout: defaultConnectTimeout,
 
-		SupportSpanmetricsConnector: false,
+		SupportSpanmetricsConnector: defaultSupportSpanmetricsConnector,
```

Comment:
This is the main (breaking) change. 

The other changes relate to local developer improvements with a similar theme of defaulting to the spanmetrics connector as well.

If we feel those changes should be decoupled into separate PRs, I'm happy to break them up.